vm/src/types/structseq.rs

```diff
@@ -0,0 +1,90 @@
+use crate::{
+    builtins::{PyTuple, PyTupleRef, PyTypeRef},
+    pyclass::{PyClassImpl, StaticType},
+    pyobject::PyContext,
+    AsObject, PyObjectRef, PyRef, PyResult, PyValue, VirtualMachine,
+};
+
+#[pyimpl]
+pub trait PyStructSequence: StaticType + PyClassImpl + Sized + 'static {
+    const FIELD_LEN: usize;
+    const FIELD_NAMES: &'static [&'static str];
```

Comment:
This is kinda redundant; is there a reason not to just do `FIELD_NAMES.len()` as before?